### PR TITLE
Added functions treepath.String() and treepath.CompactString()

### PR DESCRIPTION
**Pull request by TrevorPeters**
also added tests

### CODE REVIEW

File: core/treepath/treepath.go [hunk 1]
```diff
@@ -177,3 +177,57 @@ func (tp Treepath) Apply(n *game.Node) *game.Node {
...
+	}
+	return fmt.Sprintf("%v", strArr)
+}
+
+// CompactString returns the treepath as a CompactString (short-hand).
+// examples:
+//      []                  becomes "."
+//      [1]                 becomes ".1"
+//      [0,0,0,0]           becomes ".0:4"
+//	    [1,1,1,1]           becomes ".1:4"
```
Comment (artasparks) [inline core/treepath/treepath.go:199]: nit: weird spacing

File: core/treepath/treepath.go [hunk 2]
```diff
@@ -177,3 +177,57 @@ func (tp Treepath) Apply(n *game.Node) *game.Node {
...
+func (tp Treepath) CompactString() string {
+	var (
+		count, prev int = 1, -1
+		sb          strings.Builder
+	)
+
+	for _, v := range tp {
+
+		if v == prev {
+			//count repeated variation numbers.
```
Comment (artasparks) [inline core/treepath/treepath.go:210]: nit: typically `//<space>comment`.

File: core/treepath/treepath.go [hunk 3]
```diff
@@ -177,3 +177,57 @@ func (tp Treepath) Apply(n *game.Node) *game.Node {
...
+			//write the repeated variation number.
+			sb.WriteString(fmt.Sprintf(".%d:%d", prev, count))
+			count = 1
+		} else if prev != -1 {
+			//write non repeated variation number.
+			sb.WriteString(fmt.Sprintf(".%d", prev))
+		}
+		prev = v
+	}
+	if prev == -1 {
```
Comment (artasparks) [inline core/treepath/treepath.go:222]: I think this would be clearer with `if len(tp) == 0 {` ...

**Review by artasparks — APPROVED** [review]
Awesome work! Just a couple nits (optional to fix).

File: core/treepath/treepath.go (hunk 3 above)
Comment (TrevorPeters) [reply core/treepath/treepath.go:222]: you right

File: core/treepath/treepath.go (hunk 1 above)
Comment (TrevorPeters) [reply core/treepath/treepath.go:199]: ah, there's a tab there. whoops

**Pull request merged**